crates/symbolicator/src/server.rs

```diff
@@ -29,24 +39,46 @@ pub fn run(config: Config) -> Result<()> {
         .thread_stack_size(8 * megs)
         .build()?;
 
-    let socket = config.bind.parse::<SocketAddr>()?;
+    let mut servers: Vec<BoxFuture<_>> = vec![];
 
-    let service = web_pool
+    let service_http = web_pool
         .block_on(Service::create(
-            config,
+            config.clone(),
             io_pool.handle().to_owned(),
             cpu_pool.handle().to_owned(),
         ))
-        .context("failed to create service state")?;
+        .context("failed to create HTTP service state")?;
+    let socket_http = config.bind.parse::<SocketAddr>()?;
+    let server_http = axum_server::bind(socket_http)
+        .serve(endpoints::create_app(service_http).into_make_service());
+    servers.push(Box::pin(server_http));
+    tracing::info!("Starting HTTP server on {}", socket_http);
 
-    let _guard = web_pool.enter();
-    let server =
-        axum::Server::try_bind(&socket)?.serve(endpoints::create_app(service).into_make_service());
-
-    tracing::info!("Starting server on {}", server.local_addr());
-
-    web_pool.block_on(server)?;
+    if let Some(ref bind_str) = config.bind_https {
+        let https_conf = match config.server_config.https {
+            None => panic!("Need HTTPS config"),
+            Some(ref conf) => conf,
+        };
+        let service_https = web_pool
+            .block_on(Service::create(
+                config.clone(),
+                io_pool.handle().to_owned(),
+                cpu_pool.handle().to_owned(),
+            ))
+            .context("failed to create HTTPS service state")?;
+        let socket_https = bind_str.parse::<SocketAddr>()?;
+        let certificate = read_pem_file(&https_conf.certificate_path)?;
+        let key = read_pem_file(&https_conf.key_path)?;
+        let tls_config =
+            web_pool.block_on(async { RustlsConfig::from_pem(certificate, key).await })?;
+        let server_https = axum_server::bind_rustls(socket_https, tls_config)
+            .serve(endpoints::create_app(service_https).into_make_service());
```

Comment:
A bit similarly, I’m a bit unsure if these types are `Clone` or not. It would be nice if we could share the "endpoints" across these two servers. Though I doubt this has any negative consequences apart from maybe memory usage.

Reply:
Ah nice, I see you directly deduped the whole `endpoints::create_app(service_https).into_make_service()` expression, which is what I asked about ;-)